src/omero_cli_render.py

```diff
@@ -634,52 +685,34 @@ def set(self, args):
             else:
                 cindices.append(i)
             rangelist.append([c.start, c.end])
-            colourlist.append(c.color)
+            colorlist.append(c.color)
 
         iids = []
-        for img in self.render_images(self.gateway, args.object, batch=1):
-            iids.append(img.id)
-
-            (def_z, def_t) = self._read_default_planes(
-                img, data, ignore_errors=args.ignore_errors)
-
-            reactivatechannels = []
-            if not args.disable:
-                # Calling set_active_channels will disable channels which
-                # are not specified, have to keep track of them and
-                # re-activate them later again
-                imgchannels = img.getChannels()
-                for ci, ch in enumerate(imgchannels, 1):
-                    if ci not in cindices and -ci not in cindices\
-                            and ch.isActive():
-                        reactivatechannels.append(ci)
-
-            img.set_active_channels(
-                cindices, windows=rangelist, colors=colourlist)
-            if greyscale is not None:
-                if greyscale:
-                    img.setGreyscaleRenderingModel()
-                else:
-                    img.setColorRenderingModel()
-
-            if len(reactivatechannels) > 0:
-                img.set_active_channels(reactivatechannels)
 
-            if def_z:
-                img.setDefaultZ(def_z - 1)
-            if def_t:
-                img.setDefaultT(def_t - 1)
+        if args.batch > 1:
+            for img_batch in self.load_images(self.gateway, args.object,
+                                              batch=args.batch):
+                with concurrent.futures.ThreadPoolExecutor() as executor:
+                    future_image = {executor.submit(self._set_rend, args,
+                                                    data, img, cindices,
+                                                    greyscale, rangelist,
+                                                    colorlist):
+                                    img for img in img_batch}
+                    for future in concurrent.futures.as_completed(
+                            future_image):
+                        img = future_image[future]
+                        try:
+                            iid = future.result()
+                            iids.append(iid)
+                        except Exception as exc:
+                            print('%d generated an exception: %s' % (img.id,
```

Comment:
This should be a call to `self.ctx.err` (or possibly `ctx.out` if there's nothing on stdout that anyone might use).